api/object.go

```diff
@@ -186,14 +220,18 @@ type (
 		Limit  int
 	}
 
+	// UploadObjectOptions is the options type for the worker client.
 	UploadObjectOptions struct {
 		Offset                       int
 		MinShards                    int
 		TotalShards                  int
 		ContractSet                  string
-		MimeType                     string
 		DisablePreshardingEncryption bool
 		ContentLength                int64
+
+		// Metadata contains all object metadata and will contain things like
+		// the Content-Type as well as all user-defined metadata.
+		Metadata map[string]string
```

Comment:
Doesn't that make it annoying to use? If it's not a field then you don't know what sort of metadata you can set and how to set it. Or how to prefix it.

If the common things like `MimeType` and `ContentLength` remain fields it's a lot more straightforward imo.

`Metadata` could become `UserMetadata ObjectUserMetadata` and I'd add a `Set` and `Get` method on it as well. Then you can use it without having to know about the `X-Sia-` prefix since `Set` would implicitly add it and `Get` would implicitly remove it for you.

Reply:
Hm so having `MimeType` and `ContentLength` be fields is more straightforward. Having the `Metadata` be `ObjectUserMetadata` also makes sense. I did not add the `Get` and `Set` because I wasn't using it and because `ObjectUserMetadata` contains the keys without the prefix.